src/builder.rs

```diff
@@ -202,6 +204,15 @@ impl SessionBuilder {
         self
     }
 
+    /// Set the ControlPersist option (`ssh -o ControlPersist=<value>`).
+    ///
+    /// If not set, openssh will use "ControlPersist=yes".
+    ///
+    pub fn control_persist(&mut self, value: String) -> &mut Self {
```

Comment:
Passing in an arbitrary string is bad, user could pass `yes --...` to pass any option to the ssh.

Let's create an enum to limit possible values:

```rust
#[derive(Debug, Default)]
enum ControlPersistOptions {
    #[default]
    Forever,
    // Will be closed after the initial connection is closed
    ClosedAfterInitialConnection,
    // If the ssh control server has been idle for specified duration, it will
    // exit.
    IdleFor(Duration),
}
```